llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -257,6 +257,20 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
       .clampNumElements(0, v2s64, v2s64)
       .moreElementsToNextPow2(0);
 
+  getActionDefinitionsBuilder(G_FABS)
+      .legalFor({MinFPScalar, s32, s64, v2s32, v4s32, v2s64})
+      .legalIf([=](const LegalityQuery &Query) {
+        const auto &Ty = Query.Types[0];
+        return (Ty == v8s16 || Ty == v4s16) && HasFP16;
+      })
+      .lowerFor({s128})
+      .scalarizeIf(scalarOrEltWiderThan(0, 64), 0)
```

Comment:
Do you need the `scalarizeIf` when `lowerFAbs` supports vectors?

Reply:
Unfortunately G_AND does not yet support 128-bit types so let's keep the `scalarizeIf` with a comment.

Reply:
G_AND does not support s128 so we still need this. Left a TODO.
```
LLVM ERROR: unable to legalize instruction: %3:_(<2 x s128>) = G_AND %0:_, %7:_ (in function: fabs_v2f128)
```

Reply:
I happened to be adding i128 support for add/sub recently, and and/or/xor was caught up in them. There might not be a lot of benefit at the moment choosing one method vs the other as we will just scalarize in either case. That could be improved in the future if we change how they lower.